packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -760,8 +761,19 @@ function dispatchAction<S, A>(
 }
 
 function inputsAreEqual(arr1, arr2) {
-  // Don't bother comparing lengths because these arrays are always
+  // Don't bother comparing lengths in prod because these arrays should be
   // passed inline.
+  if (__DEV__) {
+    warning(
+      arr1.length === arr2.length,
+      'Detected a variable number of dependencies. The length of the ' +
```

Comment:
Seems like this error message should mention "hooks". It's a little cryptic.

Reply:
Perhaps "variable number of dependencies to a hook"?

Reply:
Updated

Reply:
Yeah, I like that change.